experimental/packages/otlp-exporter-base/src/platform/browser/OTLPExporterBrowserBase.ts

```diff
@@ -69,20 +64,22 @@ export abstract class OTLPExporterBrowserBase<
     const body = JSON.stringify(serviceRequest);
 
     const promise = new Promise<void>((resolve, reject) => {
-      if (this._useXHR) {
-        sendWithXhr(
+      if (!this._alwaysUseXhr && document.hidden && body.length < 64 * 1024) {
+        // sendBeacon must be used when the page is being hidden, as per: https://w3c.github.io/beacon/.
+        // Note that headers are ignored in this case, if this is a problem, configuration option `alwaysUseXhr` must be set to true.
```

Comment:
Hmm is this enough to add ` && Object.keys(this._headers).length == 0` to the if condition above? (Since it's not possible to know if a request header is required (eg. to pass auth tokens*) or optional)

(Alternatively could provide an option for user to pass their own conditional check so and have this+header len as sane default)

\* - yeah there's better alternatives but sometimes people make bad decisions

Reply:
Additionally, "if" the page is unloading then you should use (if available) the `fetch` API with the keep-alive flag enabled (this still has the 64kb limit), but you can pass headers.

Reply:
> fetch API with the keep-alive flag enabled

[MDN says that's not supported by firefox](https://developer.mozilla.org/en-US/docs/Web/API/fetch#browser_compatibility)